core/ledger/kvledger/txmgmt/txmgr/lockbased_txmgr.go

```diff
@@ -483,6 +489,12 @@ func (txmgr *LockBasedTxMgr) Shutdown() {
 	txmgr.db.Close()
 }
 
+// UpdateBatchWithAppInitiatedPvtKeysToPurge adds delete markers in the state update batch for the private data keys that
+// should be deleted from the state as a result of app initiated purge operations.
+func (txmgr *LockBasedTxMgr) UpdateBatchWithAppInitiatedPvtKeysToPurge(keys map[privacyenabledstate.PvtdataCompositeKey]*version.Height) {
+	txmgr.currentUpdates.purgeAppInitiatedPvtKeys(keys)
```

Comment:
The RFC says that purged private keys will be deleted from private state in the background. But it looks like you are deleting from private state atomically with the commit. Did you change your mind or have I got it wrong?

Reply:
Yes, your observation is correct. During the implementation, I felt that processing this in the background would have needed some changes that would make it a bit more complex. So, I picked the simplicity with the assumption that perhaps the purge is not going to be a very frequent transaction. We can always change it if we find it otherwise.